node/network/gossip-support/src/lib.rs

```diff
@@ -133,8 +121,7 @@ impl State {
 			if let Some((new_session, relay_parent)) = maybe_new_session {
 				tracing::debug!(target: LOG_TARGET, %new_session, "New session detected");
 				let validators = determine_relevant_validators(ctx, relay_parent, new_session).await?;
-				let validators = choose_random_subset(validators);
-				tracing::debug!(target: LOG_TARGET, targets = ?validators, "Issuing a connection request");
+				tracing::debug!(target: LOG_TARGET, num = ?validators.len(), "Issuing a connection request");
 
 				let request = validator_discovery::connect_to_validators_in_session(
```

Comment:
I forgot to check this - we should still make sure to only connect to all authorities if we are one ourselves. The validators are protected now, in the sense that they will be connected to each other, no matter what, but having each and every full node connect to every validator is still a bad idea:

- It is absolutely pointless
- Incoming slots of validators will all be full (of each of them) as soon as 25 full nodes join the network

A full node should only connect to a few nodes the most. Maybe just one or two picked at random. (Or none?) @rphmeier does a full node do anything useful right now on the validation peer set? E.g. would it behave as a fishermen or something on any protocol? If not, we should likely not connect at all by default.

Reply:
the problem is that at the moment it's not easy to check if we're a validator (unless we always keep `max_validators` at `None`), so I left it out for later

if we decide to never set `max_validators` though, that check can be easily added

Reply:
I think we've decided that yesterday :-) @rphmeier ? In any case we have to do something, having all full nodes connect to all validators is severely broken behaviour.

Reply:
Yeah, fair enough, I'll add a check then.